### PR TITLE
Finish up remaining documentation in iOS library

### LINKED ISSUE #983: Finish up remaining documentation
### Bug Report
While all of the public classes have documentation, some the private classes are still missing documentation. A list of those files can be found in this repo via **docs/undocumented.json**.

##### OS & Version Information
* iOS Version: n/a
* SDL iOS Version: 6.0
* Testing Against: n/a

### PR DESCRIPTION
**Pull request by justingluck93**
Fixes #983 

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
Review documentation

### Summary
Added missing documentation to classes, inits, methods, and properties

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: SmartDeviceLink/SDLArtwork.h [hunk 1]
```diff
@@ -13,13 +13,21 @@
 
 @class SDLImage;
 
+/// Format of the Artwork
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLArtwork.h]:
```suggestion
/// Image format of an artwork file
```

File: SmartDeviceLink/SDLArtwork.h [hunk 2]
```diff
@@ -13,13 +13,21 @@
...
     SDLArtworkImageFormatPNG,
+
+    /// Image format: JPG
     SDLArtworkImageFormatJPG
 };
 
 NS_ASSUME_NONNULL_BEGIN
 
+/**
+ A type of SDLFile image
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLArtwork.h]:
```suggestion
 An SDLFile subclass specifically designed for images
```

File: SmartDeviceLink/SDLAudioControlCapabilities.h [hunk 3]
```diff
@@ -6,6 +6,11 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/**
+ Current audio control capability. If included, the platform supports audio controls.
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLAudioControlCapabilities.h]:
```suggestion
 Describes a head unit's audio control capabilities.
```

File: SmartDeviceLink/SDLAudioControlData.m [hunk 4]
```diff
@@ -15,7 +15,6 @@ - (instancetype)initWithSource:(nullable SDLPrimaryAudioSource)source keepContex
     if (!self) {
         return nil;
     }
-
     self.source = source;
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLAudioControlData.m]:
```suggestion

    self.source = source;
```

Add back in newline for clarity

File: SmartDeviceLink/SDLAudioStreamManager.h [hunk 5]
```diff
@@ -16,13 +16,25 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/// Domain error for AudioStreamManager
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLAudioStreamManager.h]:
```suggestion
/// Error relates to AudioStreamManager
```

File: SmartDeviceLink/SDLChangeRegistration.h [hunk 6]
```diff
@@ -19,8 +19,28 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface SDLChangeRegistration : SDLRPCRequest
 
+/**
+Constructs a newly allocated SDLChangeRegistration object with the given parameters
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLChangeRegistration.h]:
```suggestion
Constructs a newly allocated SDLChangeRegistration object with required parameters
```

File: SmartDeviceLink/SDLChangeRegistration.h [hunk 7]
```diff
@@ -19,8 +19,28 @@ NS_ASSUME_NONNULL_BEGIN
...
+
+@param language the name of the button
+@param hmiDisplayLanguage the module where the button should be pressed
+
+@return An instance of the SDLChangeRegistration class.
+*/
 - (instancetype)initWithLanguage:(SDLLanguage)language hmiDisplayLanguage:(SDLLanguage)hmiDisplayLanguage;
 
+/**
+Constructs a newly allocated SDLChangeRegistration object with the given parameters
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLChangeRegistration.h]:
```suggestion
Constructs a newly allocated SDLChangeRegistration object with all parameters
```

File: SmartDeviceLink/SDLChoice.h [hunk 8]
```diff
@@ -9,14 +9,36 @@
...
  *
- * Since SmartDeviceLink 1.0
+ * Since RPC 1.0
  */
 NS_ASSUME_NONNULL_BEGIN
 
 @interface SDLChoice : SDLRPCStruct
 
+/**
+Constructs a newly allocated SDLChangeRegistration object with the given parameters
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLChoice.h]:
```suggestion
Constructs a newly allocated SDLChangeRegistration object with the required parameters
```

File: SmartDeviceLink/SDLChoice.h [hunk 9]
```diff
@@ -9,14 +9,36 @@
...
+@param choiceId the application-scoped identifier that uniquely identifies this choice
+@param menuName text which appears in menu, representing this choice
+@param vrCommands vr synonyms for this choice
+
+@return An instance of the SDLChangeRegistration class.
+*/
 - (instancetype)initWithId:(UInt16)choiceId menuName:(NSString *)menuName vrCommands:(nullable NSArray<NSString *> *)vrCommands;
 
+/**
+Constructs a newly allocated SDLChangeRegistration object with the given parameters
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLChoice.h]:
```suggestion
Constructs a newly allocated SDLChangeRegistration object with all parameters
```

File: SmartDeviceLink/SDLConfiguration.h [hunk 10]
```diff
@@ -17,6 +17,7 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/// Contains information about the apps configurtion, such as lifecycle, lockscreen, encryption, etc.
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLConfiguration.h]:
```suggestion
/// Contains information about the app's configurtion, such as lifecycle, lockscreen, encryption, etc.
```

File: SmartDeviceLink/SDLCreateInteractionChoiceSet.h [hunk 11]
```diff
@@ -23,6 +23,11 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface SDLCreateInteractionChoiceSet : SDLRPCRequest
 
+/// Convenience init for creating a choice set
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLCreateInteractionChoiceSet.h]:
```suggestion
/// Convenience init for creating a choice set RPC
```

File: SmartDeviceLink/SDLDeviceInfo.h [hunk 12]
```diff
@@ -10,6 +10,9 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @interface SDLDeviceInfo : SDLRPCStruct
 
+/// Convenience init. Object will contain all information about the connected device.
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLDeviceInfo.h]:
```suggestion
/// Convenience init. Object will contain all information about the connected device automatically.
```

File: SmartDeviceLink/SDLEnum.h [hunk 13]
```diff
@@ -7,8 +7,10 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/// NSString SDLEnum typedef
 typedef NSString* SDLEnum SDL_SWIFT_ENUM;
 
+/// Compares two enums for equivalents
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLEnum.h]:
```suggestion
/// Extensions to NSString specifically for SDL string enums
```

File: SmartDeviceLink/SDLFileManager.h [hunk 14]
```diff
@@ -18,6 +18,10 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/// The handler that is called when the manager is set up or failed to set up with an error
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLFileManager.h]: I think this is for internal use only, right? If so, please note that.

File: SmartDeviceLink/SDLFunctionID.h [hunk 15]
```diff
@@ -7,11 +7,22 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/// The function IDs os each RPC
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLFunctionID.h]:
```suggestion
/// A function ID for an SDL RPC
```

File: SmartDeviceLink/SDLFunctionID.h [hunk 16]
```diff
@@ -7,11 +7,22 @@
...
 @interface SDLFunctionID : NSObject
 
+/// A shared instance
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLFunctionID.h]:
```suggestion
/// The shared object for pulling function id information
```

File: SmartDeviceLink/SDLGetSystemCapability.h [hunk 17]
```diff
@@ -13,7 +13,7 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
-/*
+/**SDLGetWayPoints.
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLGetSystemCapability.h]: This doesn't look right

File: SmartDeviceLink/SDLGetWayPoints.h [hunk 18]
```diff
@@ -7,8 +7,15 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/// This RPC allows you to get waypoint data
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLGetWayPoints.h]:
```suggestion
/// This RPC allows you to get navigation waypoint data
```

File: SmartDeviceLink/SDLHybridAppPreference.h [hunk 19]
```diff
@@ -13,17 +13,17 @@
  */
 typedef SDLEnum SDLHybridAppPreference SDL_SWIFT_ENUM;
 
-/*
+/**SDLImageResolution.h
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLHybridAppPreference.h]: This doesn't look right

File: SmartDeviceLink/SDLImageResolution.h [hunk 20]
```diff
@@ -25,6 +25,11 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (strong, nonatomic) NSNumber<SDLInt> *resolutionHeight;
 
+/// Convenience init with all parametersSDLKeyboardDelegate.h
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLImageResolution.h]:
```suggestion
/// Convenience init with all parameters
```

File: SmartDeviceLink/SDLKeyboardDelegate.h [hunk 21]
```diff
@@ -35,11 +35,10 @@ typedef void(^SDLKeyboardAutoCompleteResultsHandler)(NSArray<NSString *> *_Nulla
  */
 typedef void(^SDLKeyboardCharacterSetCompletionHandler)(NSArray<NSString *> *_Nullable updatedCharacterSet);
 
+/// A protocol with available keyboard delegate methods and properties
 @protocol SDLKeyboardDelegate <NSObject>
 
-//
-
-/**
+/**SDLLifecycleConfiguration.h
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLKeyboardDelegate.h]:
```suggestion
/**
```

File: SmartDeviceLink/SDLLightControlCapabilities.h [hunk 22]
```diff
@@ -8,6 +8,10 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+
+/// Current light control capabilitie. If included, the platform supports light controls.
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLLightControlCapabilities.h]:
```suggestion
/// Current light control capabilities.
```

File: SmartDeviceLink/SDLLightControlData.h [hunk 23]
```diff
@@ -7,6 +7,9 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/// Data about the current light controls
+///
+/// @since 5.0
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLLightControlData.h]:
```suggestion
/// @since SDL 5.0
```

File: SmartDeviceLink/SDLLightState.h [hunk 24]
```diff
@@ -10,6 +10,7 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/// Current light control state
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLLightState.h]: Since when?

File: SmartDeviceLink/SDLLockScreenViewController.h [hunk 25]
```diff
@@ -10,8 +10,10 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/// The view controller fo the lockscreen.
 @interface SDLLockScreenViewController : UIViewController
 
+/// A block that can be used to close the lockscreen when the user swipes on the lockscreen.
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLLockScreenViewController.h]:
```suggestion
/// A block that can be used to close the lockscreen when the user swipes on the lockscreen. Override this in your own custom view controllers if you build a custom lock screen.
```

File: SmartDeviceLink/SDLLogConstants.h [hunk 26]
```diff
@@ -20,55 +20,70 @@ NS_ASSUME_NONNULL_BEGIN
  */
 typedef BOOL (^SDLLogFilterBlock)(SDLLogModel *log);
 
+/// An enum describing log bytes direction
 typedef NS_ENUM(NSUInteger, SDLLogBytesDirection) {
+    /// SDLLogBytesDirection
+    ///
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLLogConstants.h]:
```suggestion
```

File: SmartDeviceLink/SDLLogConstants.h [hunk 27]
```diff
@@ -20,55 +20,70 @@ NS_ASSUME_NONNULL_BEGIN
  */
 typedef BOOL (^SDLLogFilterBlock)(SDLLogModel *log);
 
+/// An enum describing log bytes direction
 typedef NS_ENUM(NSUInteger, SDLLogBytesDirection) {
+    /// SDLLogBytesDirection
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLLogConstants.h]:
```suggestion
```

File: SmartDeviceLink/SDLLogConstants.h [hunk 28]
```diff
@@ -20,55 +20,70 @@ NS_ASSUME_NONNULL_BEGIN
...
+    ///
+    /// Trasmit
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLLogConstants.h]:
```suggestion
    /// Transmit from the app
```

File: SmartDeviceLink/SDLLogConstants.h [hunk 29]
```diff
@@ -20,55 +20,70 @@ NS_ASSUME_NONNULL_BEGIN
...
     SDLLogBytesDirectionTransmit,
+
+    /// SDLLogBytesDirection
+    ///
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLLogConstants.h]:
```suggestion
```

File: SmartDeviceLink/SDLLogConstants.h [hunk 30]
```diff
@@ -20,55 +20,70 @@ NS_ASSUME_NONNULL_BEGIN
...
 typedef BOOL (^SDLLogFilterBlock)(SDLLogModel *log);
 
+/// An enum describing log bytes direction
 typedef NS_ENUM(NSUInteger, SDLLogBytesDirection) {
+    /// SDLLogBytesDirection
+    ///
+    /// Trasmit
     SDLLogBytesDirectionTransmit,
+
+    /// SDLLogBytesDirection
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLLogConstants.h]:
```suggestion
```

File: SmartDeviceLink/SDLLogConstants.h [hunk 31]
```diff
@@ -20,55 +20,70 @@ NS_ASSUME_NONNULL_BEGIN
...
+    ///
+    /// Receive
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLLogConstants.h]:
```suggestion
    /// Receive from the module
```

File: SmartDeviceLink/SDLLogFileModule.h [hunk 32]
```diff
@@ -13,6 +13,7 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/// Log file module information
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLLogFileModule.h]:
```suggestion
/// A log file module is a collection of source code files that form a cohesive unit and that logs can all use to describe themselves. E.g. a "transport" module, or a "Screen Manager" module.
```

File: SmartDeviceLink/SDLProtocolConstants.h [hunk 33]
```diff
@@ -9,74 +9,140 @@
...
  *  The data packet's format and priority.
-
- - SDLServiceTypeControl: The lowest level service available.
- - SDLServiceTypeRPC: Used to send requests, responses, and notifications between an application and a head unit.
- - SDLServiceTypeAudio: The application can start the audio service to send PCM audio data to the head unit. After the StartService packet is sent and the ACK received, the payload for the Audio Service is only PCM audio data.
- - SDLServiceTypeVideo: The application can start the video service to send H.264 video data to the head unit. After the StartService packet is sent and the ACK received, the payload for the Video Service is only H.264 video data.
- - SDLServiceTypeBulkData: Similar to the RPC Service but adds a bulk data field. The payload of a message sent via the Hybrid service consists of a Binary Header, JSON Data, and Bulk Data.
  */
 typedef NS_ENUM(UInt8, SDLServiceType) {
+    /// SDLServiceType
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLProtocolConstants.h]: Do the above for these as well

File: SmartDeviceLink/SDLReleaseInteriorVehicleDataModule.h [hunk 34]
```diff
@@ -11,8 +11,15 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/// Releases a controlled module so others can take control
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLReleaseInteriorVehicleDataModule.h]:
```suggestion
/// Releases a controlled remote control module so others can take control
```

File: SmartDeviceLink/SDLSISData.h [hunk 35]
```diff
@@ -15,6 +15,14 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface SDLSISData : SDLRPCStruct
 
+/// Convenience init to
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLSISData.h]:
```suggestion
/// Convenience init to SISData
```

File: SmartDeviceLink/SDLScreenManager.h [hunk 36]
```diff
@@ -45,6 +45,7 @@ typedef void(^SDLScreenManagerUpdateCompletionHandler)(NSError *__nullable error
  */
 typedef void(^SDLPreloadChoiceCompletionHandler)(NSError *__nullable error);
 
+/// The SDLScreenManager is a manager to control certain UI features. Use the screen manager for setting up the UI of the template, creating a menu for your users, creating softbuttons and setting textfields.
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLScreenManager.h]:
```suggestion
/// The SDLScreenManager is a manager to control SDL UI features. Use the screen manager for setting up the UI of the template, creating a menu for your users, creating softbuttons, setting textfields, etc..
```

File: SmartDeviceLink/SDLScreenManager.h [hunk 37]
```diff
@@ -242,6 +243,8 @@ If set to `SDLDynamicMenuUpdatesModeForceOff`, menu updates will work the legacy
 
 #pragma mark Soft Button
 
+/// Retrieve a SoftButtonObject from a SoftButtonObject name.
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLScreenManager.h]:
```suggestion
/// Retrieve a SoftButtonObject based on its name.
```

File: SmartDeviceLink/SDLSendLocation.h [hunk 38]
```diff
@@ -13,6 +13,9 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/// Used to create a location object for navigation
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLSendLocation.h]:
```suggestion
/// SendLocation is used to send a location to the navigation system for navigation
```

File: SmartDeviceLink/SDLSoftButton.h [hunk 39]
```diff
@@ -16,10 +16,23 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @interface SDLSoftButton : SDLRPCStruct
 
+/// Convenience init
+///
+/// @param handler A handler that may optionally be run when an SDLSubscribeButton or SDLSoftButton has a corresponding notification occur
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLSoftButton.h]:
```suggestion
/// @param handler A handler that may optionally be run when the SDLSoftButton has a corresponding notification occur
```

File: SmartDeviceLink/SDLSoftButton.h [hunk 40]
```diff
@@ -16,10 +16,23 @@ NS_ASSUME_NONNULL_BEGIN
...
 
+/// Convenience init
+///
+/// @param type Describes whether this soft button displays only text, only an image, or both
+/// @param text Optional text to display (if defined as TEXT or BOTH type)
+/// @param image Optional image struct for SoftButton (if defined as IMAGE or BOTH type)
+/// @param highlighted Displays in an alternate mode, e.g. with a colored background or foreground. Depends on the IVI system
+/// @param buttonId Value which is returned via OnButtonPress / OnButtonEvent
+/// @param systemAction Parameter indicating whether selecting a SoftButton shall call a specific system action
+/// @param handler A handler that may optionally be run when an SDLSubscribeButton or SDLSoftButton has a corresponding notification occur.
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLSoftButton.h]:
```suggestion
/// @param handler A handler that may optionally be run when the SDLSoftButton has a corresponding notification occur.
```

File: SmartDeviceLink/SDLSoftButton.h [hunk 41]
```diff
@@ -16,10 +16,23 @@ NS_ASSUME_NONNULL_BEGIN
...
 - (instancetype)initWithType:(SDLSoftButtonType)type text:(nullable NSString *)text image:(nullable SDLImage *)image highlighted:(BOOL)highlighted buttonId:(UInt16)buttonId systemAction:(nullable SDLSystemAction)systemAction handler:(nullable SDLRPCButtonNotificationHandler)handler;
 
+/// A handler that may optionally be run when an SDLSubscribeButton or SDLSoftButton has a corresponding notification occur.
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLSoftButton.h]:
```suggestion
/// A handler that may optionally be run when the SDLSoftButton has a corresponding notification occur.
```

File: SmartDeviceLink/SDLSoftButtonObject.h [hunk 42]
```diff
@@ -57,7 +58,7 @@ NS_ASSUME_NONNULL_BEGIN
 
 /**
  Create a single-state soft button. For example, a button that brings up a Perform Interaction menu.
-
+
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLSoftButtonObject.h]:
```suggestion

```

File: SmartDeviceLink/SDLSoftButtonState.h [hunk 43]
```diff
@@ -15,6 +15,7 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/// A soft button state, includes data such as text, name and artwork
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLSoftButtonState.h]:
```suggestion
/// A soft button state including data such as text, name and artwork
```

File: SmartDeviceLink/SDLSupportedSeat.h [hunk 44]
```diff
@@ -9,18 +9,18 @@
  */
 typedef SDLEnum SDLSupportedSeat SDL_SWIFT_ENUM __deprecated;
 
-/**
- * @abstract Save current seat postions and settings to seat memory.
- */
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
+/**
+* @abstract Save current seat postions and settings to seat memory.
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLSupportedSeat.h]:
```suggestion
* Save current seat positions and settings to seat memory.
```

File: SmartDeviceLink/SDLSupportedSeat.h [hunk 45]
```diff
@@ -9,18 +9,18 @@
...
 extern SDLSupportedSeat const SDLSupportedSeatDriver;
 #pragma clang diagnostic pop
 
-/**
- * @abstract Restore / apply the seat memory settings to the current seat.
- */
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
+/**
+* @abstract Restore / apply the seat memory settings to the current seat.
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLSupportedSeat.h]:
```suggestion
* Restore / apply the seat memory settings to the current seat.
```

File: SmartDeviceLink/SDLTouch.h [hunk 46]
```diff
@@ -10,13 +10,20 @@
 
 @class SDLTouchEvent;
 
+/// Enum: SDLTouchIdentifier
+///
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLTouch.h]:
```suggestion
```

File: SmartDeviceLink/SDLTouch.h [hunk 47]
```diff
@@ -10,13 +10,20 @@
 
 @class SDLTouchEvent;
 
+/// Enum: SDLTouchIdentifier
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLTouch.h]:
```suggestion
```

File: SmartDeviceLink/SDLTouchManager.h [hunk 48]
```diff
@@ -18,9 +18,13 @@
...
 NS_ASSUME_NONNULL_BEGIN
 
+/// Handler for touch events
+///
+/// @param touch Describes a touch location
+/// @param type The type of touch
 typedef void(^SDLTouchEventHandler)(SDLTouch *touch, SDLTouchType type);
 
-
+/// Touch Manager responsible for providing touch event notifications.
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLTouchManager.h]:
```suggestion
/// Touch Manager responsible for processing touch event notifications.
```

File: SmartDeviceLink/SDLVersion.h [hunk 49]
```diff
@@ -13,31 +13,107 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/// Specifies the version number of the SmartDeviceLink protocol that is supported by the mobile application.
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLVersion.h]:
```suggestion
/// Specifies a major / minor / patch version number for semantic versioning purposes and comparisons
```

File: SmartDeviceLink/SDLProtocolConstants.h (hunk 33 above)
Comment (justingluck93) [reply SmartDeviceLink/SDLProtocolConstants.h]: done!

File: SmartDeviceLink/SDLFileManager.h (hunk 14 above)
Comment (justingluck93) [reply SmartDeviceLink/SDLFileManager.h]: added

File: SmartDeviceLink/SDLHybridAppPreference.h (hunk 19 above)
Comment (justingluck93) [reply SmartDeviceLink/SDLHybridAppPreference.h]: fixed

File: SmartDeviceLink/SDLGetSystemCapability.h (hunk 17 above)
Comment (justingluck93) [reply SmartDeviceLink/SDLGetSystemCapability.h]: fixed

File: SmartDeviceLink/SDLManager.h [hunk 50]
```diff
@@ -99,12 +102,12 @@ typedef void (^SDLManagerReadyBlock)(BOOL success, NSError *_Nullable error);
...
 
-/**
- * Deprecated internal proxy object. This should only be accessed when the Manager is READY. This property may go to nil at any time.
- * The only reason to use this is to access the `putFileStream:withRequest:` method. All other functionality exists on managers in 4.3. This will be removed in 5.0 and the functionality replicated on `SDLFileManager`.
- */
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
+/**
+* Deprecated internal proxy object. This should only be accessed when the Manager is READY. This property may go to nil at any time.
+* The only reason to use this is to access the `putFileStream:withRequest:` method. All other functionality exists on managers in 4.3. This will be removed in 5.0 and the functionality replicated on `SDLFileManager`.
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLManager.h]:
```suggestion
* All the other functionality exists on managers in sdl_ios 4.3+.
```

File: SmartDeviceLink/SDLRPCMessageType.h [hunk 51]
```diff
@@ -5,13 +5,14 @@
...
- - SDLRPCMessageTypeNotification: A notification from the IVI system to the app
  */
 typedef NS_ENUM(Byte, SDLRPCMessageType) {
+    /// A request that will require a response
     SDLRPCMessageTypeRequest = 0,
+
+    /// A response to a request
     SDLRPCMessageTypeResponse,
+
+    /// A notification that does not have a response
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLRPCMessageType.h]:
```suggestion
    /// A message that does not have a response
```

File: SmartDeviceLink/SDLVersion.h [hunk 52]
```diff
@@ -13,31 +13,107 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/// Specifies a major / minor / patch version number for semantic versioning purposes and comparisons
 @interface SDLVersion : NSObject <NSCopying>
 
+/// Major version
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLVersion.h]:
```suggestion
/// Major version (e.g. X.0.0)
```

File: SmartDeviceLink/SDLVersion.h [hunk 53]
```diff
@@ -13,31 +13,107 @@
...
 @property (nonatomic, assign) NSUInteger major;
+
+/// Minor version
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLVersion.h]:
```suggestion
/// Minor version (e.g. 0.X.0)
```

File: SmartDeviceLink/SDLVersion.h [hunk 54]
```diff
@@ -13,31 +13,107 @@
...
 @property (nonatomic, assign) NSUInteger minor;
+
+/// Patch version
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLVersion.h]:
```suggestion
/// Patch version (e.g. 0.0.X)
```

**Pull request merged**